README.md

```diff
@@ -68,35 +86,58 @@ end
 ### firewall_rule
 
 #### Actions
-- `:allow`: the rule should allow incoming traffic.
-- `:deny`: the rule should deny incoming traffic.
-- `:reject`: *Default action: the rule should reject incoming traffic.
-- `:masqerade`: Add masqerade rule
-- `:redirect`: Add redirect-type rule
-- `:log`: Configure logging
-- `:remove`: Remove all rules
-
-#### Attribute Parameters
-- name: name attribute. arbitrary name to uniquely identify this firewall rule
-- protocol: valid values are: :icmp, :udp, :tcp, or protocol number. default is :tcp. Using protocol numbers is not supported using the ufw provider (default for debian/ubuntu systems).
-- port: incoming port number (ie. 22 to allow inbound SSH), or an array of incoming port numbers (ie. [80,443] to allow inbound HTTP & HTTPS). NOTE: `protocol` attribute is required with multiple ports, or a range of incoming port numbers (ie. 60000..61000 to allow inbound mobile-shell. NOTE: `protocol`, or an attribute is required with a range of ports.
-- source: ip address or subnet to filter on incoming traffic. default is `0.0.0.0/0` (ie Anywhere)
-- destination: ip address or subnet to filter on outgoing traffic.
-- dest_port: outgoing port number.
-- position: position to insert rule at. if not provided rule is inserted at the end of the rule list.
-- direction: direction of the rule. valid values are: :in, :out, default is :in
-- interface: interface to apply rule (ie. 'eth0').
-- logging: may be added to enable logging for a particular rule. valid values are: :connections, :packets. In the ufw provider, :connections logs new connections while :packets logs all packets.
-- raw: for passing a raw command to the provider (for use with custom modules, also used by zap provider to clean up non-chef managed rules)
+- `:create` (_default action_): If a firewall_rule runs this action, the rule will be recorded in a chef resource's internal state, and applied when providers automatically notify the firewall resource with action `:reload`. The notification happens automatically.
+
+#### Parameters
+
+- `firewall_name`: the matching firewall resource that this rule applies to. Default value: `default`
+
+- `raw`: Used to pass an entire rule as a string, omitting all other parameters. This line will be directly loaded by `iptables-restore`, fed directly into `ufw` on the command line, or run using `firewall-cmd`.
+
+- `description` (_default: same as rule name_): Used to provide a comment that will be included when adding the firewall rule.
+
+- `position` (_default: 50_): **relative** position to insert rule at. Position may be any integer between 0 < n < 100 (exclusive), and more than one rule may specify the same position.
```

Comment:
Hi @martinb3, not sure if you meant this by relative, but the positions are calculated based on their line in the state file of iptables (for example /etc/sysconfig/iptables), but the old behavour was that the INPUT and OUTPUT chain had their own positioning rules. this is kinda ticky where you want to insert a line on position 1 of the output chain, as you need to calculate the position of this line in the iptables state file, and do a insert on line 27 instead (if you have 26 INPUT lines before that).
numbering input and output starting at position 1, results in rules swapping places when adding/removing a rule, as the cookbook fights over who gets position 1 first ;-)